### PR TITLE
feat: QInput | Add v-on:paste event handler

### PR DESCRIPTION
**Pull request by jcperez-ch**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [X] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [X] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [X] It's submitted to the `dev` branch and _not_ the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [X] It's been tested with all Quasar themes
- [X] It's been tested on a Cordova (iOS, Android) app
- [X] It's been tested on a Electron app
- [X] Any necessary documentation has been added or updated [in the docs](https://github.com/quasarframework/quasar-framework.org/tree/dev/source) (for faster update click on "Suggest an edit on GitHub" at bottom of page) or explained in the PR's description.

If adding a **new feature**, the PR's description includes:
- [X] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
This PR is just adding the `@paste` event handler to `QInput` and `QSearch` components

### CODE REVIEW

**Comment by nothingismagick** [comment]
I think it would be good to test it on Cordova and Electron. Can you please do that for us?

File: src/mixins/input.js [hunk 1]
```diff
@@ -79,6 +79,11 @@ export default {
     __onClick (e) {
       this.focus()
       this.$emit('click', e)
+    },
+    __onPaste (e) {
+      this.focus()
+      this.$emit('paste', e)
+      this.__emit()
```
Comment (rstoenescu) [inline src/mixins/input.js]: Hi, Is this.__emit() here necessary? It actually makes the `@input/@change` be emitted twice.

File: src/mixins/input.js [hunk 2]
```diff
@@ -79,6 +79,11 @@ export default {
     __onClick (e) {
       this.focus()
       this.$emit('click', e)
+    },
+    __onPaste (e) {
+      this.focus()
```
Comment (rstoenescu) [inline src/mixins/input.js]: Why is the focus() necessary? I assume the field is already focused when user pastes in content.

**Comment by jcperez-ch** [comment]
Tested in Electron! Works good!

**Comment by nothingismagick** [comment]
I am wondering about how this could be applied to QEditor. Having a paste event registered there might also be quite ueseful. On another note, tested it in a cordova emulator?

**Comment by jcperez-ch** [comment]
@nothingismagick I tested with Android, not with iOS, is that OK?  All good.

**Comment by jcperez-ch** [comment]
Anything else to do in order to accept this PR?

**Comment by pdanpdan** [comment]
Can you please remove the focus and emit as requested?

**Comment by jcperez-ch** [comment]
Sorry @pdanpdan I did it locally, I forgot to push my changes... all good now

**Comment by nothingismagick** [comment]
@jcperez-ch You need to update your PR because the only commit here is from five days ago...

screen shot 2018-08-22 at 19 07 44

**Comment by jcperez-ch** [comment]
I know, I rebased my changes, so it retained one commit...., check the diffs

https://github.com/quasarframework/quasar/pull/2433/files
screen shot 2018-08-22 at 1 10 06 pm

**Comment by rstoenescu** [comment]
@jcperez-ch Thank you for contributing! This will get into tomorrow's new release (Quasar v0.17.10).

**Comment by jcperez-ch** [comment]
Awesome... Thanks! 💯

**Pull request merged**